pages/architecture.md

```diff
@@ -4,53 +4,50 @@ weight: 2
 ---
 
 
-These are the components that make up a fully fledged tracing system.
-
-![Architecture overview]({{ site.github.url }}/public/img/architecture-0.png)
 
 Instrumented libraries
 ----------------------
 
 Tracing information is collected on each host using the instrumented libraries
 and sent to Zipkin. When the host makes a request to another service, it passes
-a few tracing identifers along with the request so we can later tie the data
+a few tracing identifiers along with the request so we can later tie the data
 together.
 
 ![Instrumentation architecture]({{ site.github.url }}/public/img/architecture-1.png)
 
 To see if an instrumentation library already exists for your platform, see the
-list of [existing instrumentations]({{ site.github.url
-}}/pages/existing_instrumentations).
+list of [existing instrumentations]({{ site.github.url}}/pages/existing_instrumentations).
+
+![Architecture overview]({{ site.github.url }}/public/img/architecture-0.png)
 
 Transport
 ---------
 
-Spans must be transported from the services being traced to Zipkin collectors.
-There are two primary transports, Scribe and Kafka. Scribe is deprecated.
+Spans sent by the instrumented library must be transported from the services being traced to Zipkin collectors.
+There are two primary transports: Scribe and Kafka. Scribe is shown in the figure. See [Span Receivers]({{ site.github.url }}/pages/span_receivers) for more information.
+
+There are 4 components that make up Zipkin:
+* collector
+* storage
+* search
+* web UI
 
-Zipkin Collector
-----------------
+### Zipkin Collector
```

Comment:
this whole section is out of date. 
Let's talk about components, not daemons as the latter is an implementation detail.

Here's a deck which has some of this.. https://speakerdeck.com/adriancole/how-to-properly-blame-things-for-causing-latency-voxxed-berlin

the components are instrumentation, transport, storage, api, and ui

instrumentation is in the app, ex a library similar to a logger. The simplest zipkin topology of the remaining components are one server process, running the transport, api, and the UI (ex. zipkin-java's zipkin-service or zipkin-query-service). Other topologies are possible, but composed of the same components.

here's some notes: 

instrumentation lives in apps. It propagates trace identifiers downstream so that the next hop knows whether it is in an existing trace or not. It sends trace details (spans) out-of-band to zipkin over a transport like http or kafka

zipkin receives spans from instrumentation on a transport like http or kafka and writes them to storage, such as mysql or cassandra

the zipkin api primarily serves the javascript web ui, and it queries storage based on criteria such as service name.

from a deployment pov, instrumentation live in apps being traced (or a delegate for them).

the transport can be intermediated with a queue like kafka, or be a direct connection, like http

the api and ui are often cohosted on the same process, but for example, the javascript assets (zipkin-ui) could be hosted on nginx.